generate/generate.go

```diff
@@ -42,3 +66,275 @@ func NewResourceGenerator(urls []string) (*ResourceGenerator, error) {
 	return rg, nil
 
 }
+
+// Generate generates Go structs and a JSON Schema from the AWS CloudFormation
+// Resource Specifications provided to NewResourceGenerator
+func (rg *ResourceGenerator) Generate() error {
+
+	for name, uri := range rg.urls {
+		fmt.Printf("Downloading %s specification from %s\n", name, uri)
+		data, err := rg.downloadSpec(uri)
+		if err != nil {
+			return err
+		}
+		spec, err := rg.processSpec(name, data)
+		if err != nil {
+			return err
+		}
+		if err := rg.generateJSONSchema(name, spec); err != nil {
+			return err
+		}
+	}
+
+	for name, filename := range rg.filenames {
+		fmt.Printf("Downloading %s specification from %s\n", name, filename)
+		data, err := ioutil.ReadFile(strings.Replace(filename, "file://", "", -1))
+		if err != nil {
+			return err
+		}
+		spec, err := rg.processSpec(name, data)
```

Comment:
You could merge this loop and the previous one if you move the download/ioRead to a separate method